### DISCUSSION TITLE
Inbound Shipment: Add Item

### DISCUSSION THREAD

**Discussion by mark-prins** [openmsupply/openmsupply-client#481] [Show and tell]
The inbound shipment is using tabs to break up the large number of columns, and has the ability to add a new row for each batch that you'd like to request, using this small circle button:

<img width="1085" alt="Screen Shot 2021-11-24 at 10 24 49 AM" src="https://user-images.githubusercontent.com/9192912/143131557-64be8866-bb37-48fd-b42a-2892f255c44e.png">

Once a row has been added you can change tabs to see different columns:

<img width="1057" alt="Screen Shot 2021-11-24 at 10 23 56 AM" src="https://user-images.githubusercontent.com/9192912/143131463-49168dce-4ce7-4765-9042-efc5aa927dee.png">

<img width="1057" alt="Screen Shot 2021-11-24 at 10 27 59 AM" src="https://user-images.githubusercontent.com/9192912/143132001-c5b9fd78-9159-401d-a50b-807e1c468f07.png">

Thoughts? Would you prefer all columns to be visible, with a horizontal scroll? One row per item only?

**Comment by richardmoizeau** [comment]
**Discrepancies Management:**

1. When supplier is not using mSupply (ad hoc Inbound shipment): 

I like the tabs and the overall design. I'm just not quite sure to understand how to manage the discrepancies. In the 'Discrepancies' tab, you can encode: batch, unit quantity, Invoice nb of pack, invoice pack size, Invoice unit quantity. 

If we imagine the following case: a user receive a package of 10 pack of Paracetamol (100 units/pack), 2 different batch numbers + a paper invoice. **Supplier is not using mSupply so user will need to create an adhoc Inbound Shipments**. Unfortunately there are discrepancies between the invoice and the actual delivery:
- Invoice says 10 packs of Paracetamol (100 units/pack) but there are only 9 packs in the package
- invoice says there is only 1 batch (ABC123/Exp. Date: 12/12/2021) but there are actually 2 in the package: 5 packs with batch ABC123 and 4 packs with batch DEF456/exp date: 12/01/2022. 

How the user would be able to keep track of these discrepancies ? 

2. Supplier is using mSupply (sync'd Inbound Shipment with status = ### Shipped): 
 
Same case but this time, customer received a package and will check in mSupply is there is a matching Inbound Shipment. Same discrepancies. 

How the user would be able to keep track of these discrepancies ? 

Thanks.

**Reply by josh-griffin** [reply]
Put there because mSupply has invoice quantity and invoice pack size. 

The batch is actually the same in all tabs. Maybe it should only be editable in the `Batch` tab?

I guess the user could add two lines:
- line1: Batch ABC123, number of packs 5, invoice quantity is 10
- line2: Batch DEF456, number of packs 4, invoice quantity is 0 - you would technically lose the information that this was technically a replacement for the lack of ABC123. msupply doesn't currently cover this so wasn't trying to cover it. 

Could use a 'reason' option or a line comment as a possible solution. Would anyone be wanting to run a report to determine how often this happens or something?

**Reply by richardmoizeau** [reply]
Understood. So in the case of an adhoc Inb Shipment, user will enter in the 1st tab ('Batch') what he actually received and he will go to the 'Discrepancies' tab to enter what was in the invoice. I like the idea that he only have to do it if there are discrepancies. 

If the inbound shipment was received thru sync, then the user will update quantities and batches in the 1st tab ('Batch') and could the system automatically populate the discrepancies ?

**Reply by richardmoizeau** [reply]
> Could use a 'reason' option or a line comment as a possible solution. Would anyone be wanting to run a report to determine how often this happens or something?

I think we only need to keep track of the discrepancies. The most important here is to know that we have 9 packs i/o 10 and that 4 packs have a different batch numbers. saying that the 'new' batch is a replacement is an interpretation. Could also be an error of the supplier's warehouse that did not pick up the right batch.

**Reply by josh-griffin** [reply]
> If the inbound shipment was received thru sync, then the user will update quantities and batches in the 1st tab ('Batch') and could the system automatically populate the discrepancies ?

Yup 👍

**Comment by richardmoizeau** [comment]
**Adding custom stock fields (eg. Donor):** 

It's an open question, I don't have the answer. Just wandered if we should allow user to enter a custom stock field (eg. Donor) at the batch level ? Wonder if there could a different donor for 2 batches of the same item... Will chat with Craig about it. I guess the solution would be add the same item twice with a different donor. Would that be possible or would the system prevent it ?
